subsys/bluetooth/mesh/rpl_storage_settings.c

```diff
@@ -0,0 +1,279 @@
+		if (!entry) {
+			BT_ERR("Unable to allocate RPL entry for 0x%04x", src);
+			return -ENOMEM;
+		}
+	}
+
+	err = bt_mesh_settings_set(read_cb, cb_arg, &rpl, sizeof(rpl));
+	if (err) {
+		BT_ERR("Failed to set `net`");
+		return err;
+	}
+
+	entry->seq = rpl.seq;
+	entry->old_iv = rpl.old_iv;
+
+	BT_DBG("RPL entry for 0x%04x: Seq 0x%06x old_iv %u", entry->src,
+	       entry->seq, entry->old_iv);
+
+	return 0;
+}
+
+BT_MESH_SETTINGS_DEFINE(rpl, "RPL", rpl_set);
+
+static void store_rpl(struct bt_mesh_rpl *entry)
+{
+	struct rpl_val rpl;
+	char path[18];
+	int err;
+
+	if (!entry->src) {
+		return;
+	}
+
+	BT_DBG("src 0x%04x seq 0x%06x old_iv %u", entry->src, entry->seq,
+	       entry->old_iv);
+
+	rpl.seq = entry->seq;
+	rpl.old_iv = entry->old_iv;
+
+	snprintk(path, sizeof(path), "bt/mesh/RPL/%x", entry->src);
+
+	err = settings_save_one(path, &rpl, sizeof(rpl));
+	if (err) {
+		BT_ERR("Failed to store RPL %s value", log_strdup(path));
+	} else {
+		BT_DBG("Stored RPL %s value", log_strdup(path));
+	}
+}
+
+static void store_pending_rpl(struct bt_mesh_rpl *rpl)
+{
+	BT_DBG("");
+
+	if (atomic_test_and_clear_bit(store, rpl_idx(rpl))) {
+		store_rpl(rpl);
+	}
+}
+
+void bt_mesh_rpl_pending_store(uint16_t addr)
+{
+	int i;
+
+	if (!BT_MESH_ADDR_IS_UNICAST(addr) &&
+	     addr != BT_MESH_ADDR_ALL_NODES) {
+		return;
+	}
+
+	if (addr == BT_MESH_ADDR_ALL_NODES) {
+		bt_mesh_settings_store_cancel(BT_MESH_SETTINGS_RPL_PENDING);
+	}
+
+	for (i = 0; i < CONFIG_BT_MESH_CRPL; i++) {
+		if (addr != BT_MESH_ADDR_ALL_NODES &&
+		    addr != replay_list[i].src) {
+			continue;
+		}
+
+		if (atomic_test_bit(bt_mesh.flags, BT_MESH_VALID)) {
+			store_pending_rpl(&replay_list[i]);
+		} else {
+			clear_rpl(&replay_list[i]);
+		}
+
+		if (addr != BT_MESH_ADDR_ALL_NODES) {
+			break;
+		}
+	}
+}
+
+void bt_mesh_rpl_strg_store(struct bt_mesh_rpl *rpl, bool force)
+{
+	if (IS_ENABLED(CONFIG_BT_SETTINGS)) {
+		schedule_rpl_store(rpl, true);
+	}
+}
+
+void bt_mesh_rpl_strg_clear(struct bt_mesh_rpl *rpl)
+{
+	if (IS_ENABLED(CONFIG_BT_SETTINGS)) {
+		clear_rpl(rpl);
+	} else {
+		(void)memset(rpl, 0, sizeof(*rpl));
+	}
+}
+
+void bt_mesh_rpl_strg_clear_all(void)
+{
+	if (IS_ENABLED(CONFIG_BT_SETTINGS)) {
+		bt_mesh_settings_store_schedule(BT_MESH_SETTINGS_RPL_PENDING);
+	} else {
+		(void)memset(replay_list, 0, sizeof(replay_list));
+	}
+}
+
+int bt_mesh_rpl_get(uint16_t addr, struct bt_mesh_rpl **rpl)
+{
+	for (int i = 0; i < ARRAY_SIZE(replay_list); i++) {
+		if (!replay_list[i].src || replay_list[i].src == addr) {
+			*rpl = &replay_list[i];
+			return 0;
+		}
+	}
+
+	return -ENOMEM;
+}
+
+struct bt_mesh_rpl *bt_mesh_rpl_next(struct bt_mesh_rpl *curr)
+{
+	int i = curr ? rpl_idx(curr) + 1 : 0;
+
+	if ((i < 0) || (i >= ARRAY_SIZE(replay_list) - 1)) {
+		BT_ERR("Invalid index for input RPL entry");
+		return NULL;
+	}
+
+	for (; i < ARRAY_SIZE(replay_list); i++) {
+		if (replay_list[i].src) {
+			return &replay_list[i];
+		}
+	}
+
+	return NULL;
```

Comment:
```suggestion
    struct bt_mesh_rpl *rpl = curr ? curr + 1 : &replay_list[0];
    return rpl < &replay_list[CONFIG_BT_MESH_CRPL] ? rpl : NULL;
```

Reply:
```suggestion
	for (int i = curr ? rpl_idx(curr) + 1 : 0; i < ARRAY_SIZE(replay_list); i++) {
		if (replay_list[i].src) {
			return &replay_list[i];
		}
	}

	return NULL;
```
Please disregard the first suggestion. I forgot about address check. Simply there is no need in if-statement if I see correctly.